### PR TITLE
avoid nested if statements

### PR DESCRIPTION
**Pull request by carlosp420**

### CODE REVIEW

**Comment by pablohoffman** [comment]
Logic is not preserved

**Comment by carlosp420** [comment]
thanks for feedback Pablo. It took me a while to realize my error. I even had to write test cases to figure it out.
Maybe the test cases can be useful.

**Comment by pablohoffman** [comment]
No problem. Sorry for the succinctness of the message, but I was on my phone.

**Comment by jdemaeyer** [comment]
Test cases are always useful :)

I like splitting up this function, I'm tired of PyLint complaining about its complexity. Maybe instead of outsourcing the "If deploy successful" part we could outsource the whole "handle deploy response" part (i.e. the `with NamedTemporaryFile` block)? I think it makes more sense as a self-contained unit.

File: tests/test_utils.py [hunk 1]
```diff
@@ -258,6 +263,30 @@ class MockException(Exception):
         with self.assertRaises(MockException):
             utils.update_available(silent_fail=False)
 
+    @patch('shub.utils._is_deploy_successful', return_value=True)
+    def test_echo_short_log_if_deployed_true(self, mock_dep_success):
+        log_file = LogFile()
```
Comment (jdemaeyer) [inline tests/test_utils.py]: You could also use a `mock.Mock` instance. Then you wouldn't have to define your own class

File: tests/test_utils.py [hunk 2]
```diff
@@ -258,6 +263,30 @@ class MockException(Exception):
...
+        verbose = True
+        utils.echo_short_log_if_deployed(last_logs, log_file, verbose)
+        self.assertEqual(None, log_file.delete)
+
+        verbose = False
+        utils.echo_short_log_if_deployed(last_logs, log_file, verbose)
+        self.assertEqual(None, log_file.delete)
+
+    @patch('shub.utils._is_deploy_successful', return_value=None)
+    def test_echo_short_log_if_deployed_none(self, mock_dep_success):
```
Comment (jdemaeyer) [inline tests/test_utils.py]: These two tests have very similar overhead. Maybe there's a way to unify them? (You can change the return value during the test by setting `mock_dep_sucess.return_value`)

File: tests/test_utils.py (hunk 1 above)
Comment (carlosp420) [inline tests/test_utils.py]: ok!

**Comment by carlosp420** [comment]
Hi @jdemaeyer. I extracted the section you mentioned and added a test.

**Comment by dangra** [comment]
@carlosp420 can you rebase?

**Comment by carlosp420** [comment]
done! @dangra

**Comment by dangra** [comment]
LGTM. Leaving the final call to @jdemaeyer

File: tests/test_utils.py [hunk 3]
```diff
@@ -337,6 +337,37 @@ def _call(*args, **kwargs):
...
         self.assertIn('--no-binary=:all:', pipargs)
 
+    @patch('shub.utils._is_deploy_successful')
+    def test_echo_short_log_if_deployed(self, mock_dep_success):
+        log_file = Mock(delete=None)
+        last_logs = ["last log line"]
+
+        mock_dep_success.return_value = True
+        for verbose in [True, False]:
+            if verbose:
```
Comment (jdemaeyer) [inline tests/test_utils.py]: Hm, why are the `if` and `else` branches identical?

Comment (dangra) [inline tests/test_utils.py]: funny,  probably caused by rebase :)

Comment (carlosp420) [inline tests/test_utils.py]: doh! Fixed.

**Comment by jdemaeyer** [comment]
Thanks @carlosp420

**Pull request merged**